src/company_cmd.cpp

```diff
@@ -356,17 +356,14 @@ CommandCost CheckTileOwnership(TileIndex tile)
  */
 static void GenerateCompanyName(Company *c)
 {
-	/* Reserve space for extra unicode character. We need to do this to be able
-	 * to detect too long company name. */
-	char buffer[(MAX_LENGTH_COMPANY_NAME_CHARS + 1) * MAX_CHAR_LENGTH];
-
 	if (c->name_1 != STR_SV_UNNAMED) return;
 	if (c->last_build_coordinate == 0) return;
 
 	Town *t = ClosestTownFromTile(c->last_build_coordinate, UINT_MAX);
 
 	StringID str;
 	uint32 strp;
+	std::string name;
```

Comment:
if i'm not mistaken, this doesn't need to be declared anywhere beyond where it's used

Reply:
My compiler disagrees / complains otherwise:
```
src/company_cmd.cpp: In function ‘void GenerateCompanyName(Company*)’:
src/company_cmd.cpp:379:1: error: jump to label ‘set_name’ [-fpermissive]
  379 | set_name:;
      | ^~~~~~~~
src/company_cmd.cpp:400:8: note:   from here
  400 |   goto set_name;
      |        ^~~~~~~~
src/company_cmd.cpp:376:15: note:   skips initialization of ‘std::string name’
  376 |   std::string name = GetString(str);
      |               ^~~~
```

I do agree that it shouldn't be needed, but that would require rewriting the function to not use `goto`. That's another battle to fight somewhere in the future.

Reply:
Ah. Lol.